nav2_costmap_2d/test/integration/inflation_tests.cpp

```diff
@@ -146,22 +154,22 @@ void TestNode::validatePointInflation(
         if (cell.x_ > 0) {
           CellData data(costmap->getIndex(cell.x_ - 1, cell.y_),
             cell.x_ - 1, cell.y_, cell.src_x_, cell.src_y_);
-          m[dist].push_back(data);
+          m[distanceFromSource(cell)].push_back(data);
```

Comment:
Can't this just be `dist` you create in 145? I think these 4 can all just be `dist` like original since you updated the way its computed in 145.

Reply:
Technically they are supposed to be different, for example if we use the same `dist` from 145 then the distance of the immediate neighbors will be put inserted at 0.0 in the map whereas if we recompute `dist` before enqueuing then they will be inserted at 1.0.

In inflation layer implementation it is computed just before enqueuing.
https://github.com/ros-planning/navigation2/blob/6fb43d4a0543897d29d38b3e4a749b6bea6108e1/nav2_costmap_2d/plugins/inflation_layer.cpp#L282-L291

Reply:
https://github.com/ros-planning/navigation2/blob/fd1fef36571f221144c2e7e36dcef4cd67173bb6/nav2_costmap_2d/test/integration/inflation_tests.cpp#L145

You set dist, which is a function of cell, which doesn't change over the body of the for loop. You don't override the value of `dist`. If you hit this body https://github.com/ros-planning/navigation2/blob/fd1fef36571f221144c2e7e36dcef4cd67173bb6/nav2_costmap_2d/test/integration/inflation_tests.cpp#L157-L172 then all 4 of these additional calls are exactly equal to `dist`. 

`cell` doesnt change, so `dist` doesn't change.

Reply:
Please point out if I'm missing something but this looks awfully constant to me.

Reply:
Oooops, sorry missed it. Must have been drowsy when i wrote this :sweat_smile: 

I meant `   m[distanceFromSource(data)].push_back(data);